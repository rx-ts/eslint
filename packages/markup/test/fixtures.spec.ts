import path from 'path'

import { ESLint } from 'eslint'

const eslint = new ESLint({
  useEslintrc: false,
  ignore: false,
  overrideConfig: {
    overrides: [
      {
        files: '*.html',
        extends: 'plugin:markup/recommended',
      },
    ],
  },
})

test('fixtures', async () => {
  const results = await eslint.lintFiles(
    path.resolve(__dirname, 'fixtures/*.html'),
  )
  for (const { filePath, messages } of results) {
    expect(messages).toMatchSnapshot(path.basename(filePath))
  }
})
