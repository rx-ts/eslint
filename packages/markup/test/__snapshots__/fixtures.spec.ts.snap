// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures: attr-equal-space-after.html 1`] = `
Array [
  Object {
    "column": 10,
    "fix": Object {
      "range": Array [
        0,
        106,
      ],
      "text": "<img src=\\"path/to\\">
<img src =\\"path/to\\">
<img
    src=\\"path/to\\">
<img
    src
    =\\"path/to\\">
",
    },
    "line": 1,
    "message": "Never insert space after equal sign of attribute",
    "nodeType": null,
    "ruleId": "markup/attr-equal-space-after",
    "severity": 1,
  },
  Object {
    "column": 10,
    "line": 2,
    "message": "Never insert space after equal sign of attribute",
    "nodeType": null,
    "ruleId": "markup/attr-equal-space-after",
    "severity": 1,
  },
  Object {
    "column": 9,
    "line": 4,
    "message": "Never insert space after equal sign of attribute",
    "nodeType": null,
    "ruleId": "markup/attr-equal-space-after",
    "severity": 1,
  },
  Object {
    "column": 9,
    "line": 7,
    "message": "Never insert space after equal sign of attribute",
    "nodeType": null,
    "ruleId": "markup/attr-equal-space-after",
    "severity": 1,
  },
]
`;

exports[`fixtures: attr-spacing.html 1`] = `
Array [
  Object {
    "column": 20,
    "fix": Object {
      "range": Array [
        0,
        34,
      ],
      "text": "<img src=\\"path/to\\" src=\\"path/to2\\">
",
    },
    "line": 1,
    "message": "Required space",
    "nodeType": null,
    "ruleId": "markup/attr-spacing",
    "severity": 1,
  },
]
`;

exports[`fixtures: combine.html 1`] = `
Array [
  Object {
    "column": 10,
    "fix": Object {
      "range": Array [
        0,
        141,
      ],
      "text": "<img src=\\"path/to\\">
<img src =\\"path/to\\">
<img
    src=\\"path/to\\">
<img
    src
    =\\"path/to\\">

<img src=\\"path/to\\" src=\\"path/to2\\">
",
    },
    "line": 1,
    "message": "Never insert space after equal sign of attribute",
    "nodeType": null,
    "ruleId": "markup/attr-equal-space-after",
    "severity": 1,
  },
  Object {
    "column": 10,
    "line": 2,
    "message": "Never insert space after equal sign of attribute",
    "nodeType": null,
    "ruleId": "markup/attr-equal-space-after",
    "severity": 1,
  },
  Object {
    "column": 9,
    "line": 4,
    "message": "Never insert space after equal sign of attribute",
    "nodeType": null,
    "ruleId": "markup/attr-equal-space-after",
    "severity": 1,
  },
  Object {
    "column": 9,
    "line": 7,
    "message": "Never insert space after equal sign of attribute",
    "nodeType": null,
    "ruleId": "markup/attr-equal-space-after",
    "severity": 1,
  },
  Object {
    "column": 20,
    "line": 11,
    "message": "Required space",
    "nodeType": null,
    "ruleId": "markup/attr-spacing",
    "severity": 1,
  },
]
`;

exports[`fixtures: doctype.html 1`] = `
Array [
  Object {
    "column": 2,
    "line": 1,
    "message": "Required doctype",
    "nodeType": null,
    "ruleId": "markup/doctype",
    "severity": 2,
  },
]
`;

exports[`fixtures: empty.html 1`] = `Array []`;
